src/core/net/dns_headers.cpp

```diff
@@ -316,19 +373,59 @@ otError Name::LabelIterator::ReadLabel(char *aLabelBuffer, uint8_t &aLabelLength
     return error;
 }
 
-void ResourceRecord::Init(uint16_t aType, uint16_t aClass, uint32_t aTtl)
+const char *ResourceRecord::TypeToString(uint16_t aType)
 {
-    SetType(aType);
-    SetClass(aClass);
-    SetTtl(aTtl);
-    SetLength(0);
+    const char *str = "Unknown";
+
+    switch (aType)
+    {
+    case kTypeZero:
+        str = "ZERO";
+        break;
+    case kTypeA:
+        str = "A";
+        break;
+    case kTypeSoa:
+        str = "SOA";
+        break;
+    case kTypePtr:
+        str = "PTR";
+        break;
+    case kTypeTxt:
+        str = "TXT";
+        break;
+    case kTypeSig:
+        str = "SIG";
+        break;
+    case kTypeKey:
+        str = "KEY";
+        break;
+    case kTypeAaaa:
+        str = "AAAA";
+        break;
+    case kTypeSrv:
+        str = "SRV";
+        break;
+    case kTypeOpt:
+        str = "OPT";
+        break;
+    case kTypeAny:
+        str = "ANY";
+        break;
+    default:
+        break;
+    };
+
+    return str;
 }
 
-void ResourceRecordAaaa::Init(void)
+void UpdateLeaseOptRecord::Init(void)
 {
-    ResourceRecord::Init(kTypeAaaa);
-    SetLength(sizeof(mAddress));
-    mAddress.Clear();
+    OptRecord::Init(kClassZero); // Update lease uses CLASS value zero.
+    SetTtl(0);
```

Comment:
[RFC 6891 Section 6.1.2](https://tools.ietf.org/html/rfc6891#section-6.1.2) specifies non-zero values for CLASS and TTL.

Reply:
Yes. That's correct for general OPT RR, but https://tools.ietf.org/html/draft-sekar-dns-ul-02#section-4 which defines the "Update Lease" OPT (and is referenced by SRP spec), seem to set the the CLASS and TTL as zero.

The `UpdateLeaseOptRecord` type is only for this specific "Update Lease" OPT RR.

Reply:
@Abhayakara has a different opinion in https://github.com/Abhayakara/mdnsresponder/issues/5#issuecomment-754711228:

> draft-sekar is wrong (mostly in the sense of specifying something that is already specified in detail elsewhere). Take a look at RFC6891

Reply:
In case it's not clear from the comment on the PR, it's simply out of scope for EDNS(0) options to specify what goes in the QCLASS and TTL fields, because those fields have values that are specific to EDNS(0) itself. There should only be one OPT RR, which may contain many EDNS(0) options; indeed the latest srp-client code sends two different options. So if different EDNS(0) options specify different values for QCLASS and/or TTL, there's no way to resolve the conflict. As for conflicts with EDNS(0), an example would be that, you can't set DO=1 if you believe draft-sekar, which means that draft-sekar, if followed, breaks DNSSEC.

Reply:
I see. It makes sense. So we will go with what is specified in RFC 6891.

Reply:
In the latest push, updated the code to follow RFC 6891 related to OPT RR.